element_array_ephys/ephys.py

```diff
@@ -0,0 +1,746 @@
+        spike_depths : longblob  # (um) array of depths associated with each spike, relative to the (0, 0) of the probe    
+        """
+
+    def make(self, key):
+        output_dir = (Curation & key).fetch1('curation_output_dir')
+        kilosort_dir = find_full_path(get_ephys_root_data_dir(), output_dir)
+
+        kilosort_dataset = kilosort.Kilosort(kilosort_dir)
+        acq_software = (EphysRecording & key).fetch1('acq_software')
+
+        # ---------- Unit ----------
+        # -- Remove 0-spike units
+        withspike_idx = [i for i, u in enumerate(kilosort_dataset.data['cluster_ids'])
+                         if (kilosort_dataset.data['spike_clusters'] == u).any()]
+        valid_units = kilosort_dataset.data['cluster_ids'][withspike_idx]
+        valid_unit_labels = kilosort_dataset.data['cluster_groups'][withspike_idx]
+        # -- Get channel and electrode-site mapping
+        channel2electrodes = get_neuropixels_channel2electrode_map(key, acq_software)
+
+        # -- Spike-times --
+        # spike_times_sec_adj > spike_times_sec > spike_times
+        spike_time_key = ('spike_times_sec_adj' if 'spike_times_sec_adj' in kilosort_dataset.data
+                          else 'spike_times_sec' if 'spike_times_sec'
+                                                    in kilosort_dataset.data else 'spike_times')
+        spike_times = kilosort_dataset.data[spike_time_key]
+        kilosort_dataset.extract_spike_depths()
+
+        # -- Spike-sites and Spike-depths --
+        spike_sites = np.array([channel2electrodes[s]['electrode']
+                                for s in kilosort_dataset.data['spike_sites']])
+        spike_depths = kilosort_dataset.data['spike_depths']
+
+        # -- Insert unit, label, peak-chn
+        units = []
+        for unit, unit_lbl in zip(valid_units, valid_unit_labels):
+            if (kilosort_dataset.data['spike_clusters'] == unit).any():
+                unit_channel, _ = kilosort_dataset.get_best_channel(unit)
+                unit_spike_times = (spike_times[kilosort_dataset.data['spike_clusters'] == unit]
+                                    / kilosort_dataset.data['params']['sample_rate'])
+                spike_count = len(unit_spike_times)
+
+                units.append({
+                    'unit': unit,
+                    'cluster_quality_label': unit_lbl,
+                    **channel2electrodes[unit_channel],
+                    'spike_times': unit_spike_times,
+                    'spike_count': spike_count,
+                    'spike_sites': spike_sites[kilosort_dataset.data['spike_clusters'] == unit],
+                    'spike_depths': spike_depths[kilosort_dataset.data['spike_clusters'] == unit]})
+
+        self.insert1(key)
+        self.Unit.insert([{**key, **u} for u in units])
+
+
+@schema
+class WaveformSet(dj.Imported):
+    definition = """
+    -> CuratedClustering
+    """
+
+    class PeakWaveform(dj.Part):
```

Comment:
I hate to suggest name changes at this point, but when reading, it feels like `UnitPeakWaveform` is a better one.